author_postprocessing/author_postprocessing.py

```diff
@@ -0,0 +1,233 @@
+file of the project (authors, commits, emails, issues, etc.)
+
+If more than two person identities should be mapped to each other, several lines are necessary in the disambiguation
+file. E.g., if persons A, B, C should be mapped to A, there has to be a line which replaces B by A (A,B) and a line
+which replaces C by A (A,C).
+"""
+
+import argparse
+import sys
+from os import path, walk
+from os.path import abspath
+
+from codeface.cli import log
+from codeface.configuration import Configuration
+from codeface.dbmanager import DBManager
+
+from csv_writer import csv_writer
+
+
+##
+# RUN POSTPROCESSING
+##
+
+
+def run_postprocessing(conf, resdir):
+    """
+    Runs the postprocessing for the given parameters, that is, read the disambiguation file of the project
+    and replace all author names and e-mail addresses in all other .list files according to the disambiguation file
+
+    :param conf: the Codeface configuration object
+    :param resdir: the Codeface results dir, where output files are written
+    """
+
+    log.info("%s: Postprocess authors after manual disambiguation" % conf["project"])
+
+    authors_list = "authors.list"
+    commits_list = "commits.list"
+    emails_list = "emails.list"
+    issues_github_list = "issues-github.lisst"
+    issues_jira_list = "issues-jira.list"
+    bugs_jira_list = "bugs-jira.list"
+
+    disambiguation_list = path.join(resdir, conf["project"], conf["tagging"], "disambiguation-after-db.list")
+
+    # Check if a disambiguation list exists - if not, just stop
+    if path.exists(disambiguation_list):
+        disambiguation_data = csv_writer.read_from_csv(disambiguation_list)
+    else:
+        log.info("Disambiguation file does not exist: %s", disambiguation_list)
+        log.info("No postprocessing performed!")
+        return
+
+    # Check for all files in the result directory of the project whether they need to be adjusted
+    for filepath, dirnames, filenames in walk(path.join(resdir, conf["project"], conf["tagging"])):
+        
+        # (1) Adjust authors lists
+        if authors_list in filenames:
+            f = path.join(filepath, authors_list)
+            log.info("Postprocess %s ...", f)
+            authors = csv_writer.read_from_csv(f)
+
+            authors_to_remove = []
+            authors_new = []
+
+            # get persons which should be removed
+            for person in disambiguation_data:
+                authors_to_remove.append([person[3], person[4], person[5]])
+
+            for author in authors:
+                # keep author entry only if it should not be removed
+                if not author in authors_to_remove:
+                    authors_new.append(author)
+            csv_writer.write_to_csv(f, authors_new)
+
+        # (2) Adjust commits lists
+        if commits_list in filenames:
+            f = path.join(filepath, commits_list)
+            log.info("Postprocess %s ...", f)
+            authors = csv_writer.read_from_csv(f)
```

Comment:
Good point. I was just too lazy to do that.
In the beginning, I just tested the script on the `authors.list` data and, therefore, I named this variable `author`. Afterwards, I applied this to all the other files via copy&paste, not changing the variable name because of possible later function extractions. Then I recognized that function extraction does not make sense due to too different columns accessed, but I was too lazy to rename the variables. But you are completely right, the variables should definitively be renamed :+1: